src/main/java/org/elasticsearch/index/query/AndFilterParser.java

```diff
@@ -114,7 +116,11 @@ public Filter parse(QueryParseContext parseContext) throws IOException, QueryPar
         }
 
         // no need to cache this one
-        Filter filter = new AndFilter(filters);
+        BooleanQuery boolQuery = new BooleanQuery();
+        for (Filter filter : filters) {
+            boolQuery.add(filter, Occur.FILTER);
```

Comment:
I don't understand the need to use Occur.FILTER here (or in other parsers) versus Occur.MUST. To me these are just parsers for queries, and thats the correct logic, the fact it is wrapped as a Filter means these will all become non-scoring clauses.